easybuild/easyblocks/b/boost.py

```diff
@@ -167,20 +167,26 @@ def configure_step(self):
 
             write_file('user-config.jam', txt, append=True)
 
-    def build_boost_variant(self, bjamoptions, paracmd):
+    def build_boost_variant(self, bjamcmd, bjamoptions, paracmd):
         """Build Boost library with specified options for bjam."""
         # build with specified options
-        cmd = "%s ./bjam %s %s %s" % (self.cfg['prebuildopts'], bjamoptions, paracmd, self.cfg['buildopts'])
+        cmd = "%s ./%s %s %s %s" % (self.cfg['prebuildopts'], bjamcmd, bjamoptions, paracmd, self.cfg['buildopts'])
         run_cmd(cmd, log_all=True, simple=True)
         # install built Boost library
-        cmd = "%s ./bjam %s install %s %s" % (self.cfg['preinstallopts'], bjamoptions, paracmd, self.cfg['installopts'])
+        cmd = "%s ./%s %s install %s %s" % (
+            self.cfg['preinstallopts'], bjamcmd, bjamoptions, paracmd, self.cfg['installopts'])
         run_cmd(cmd, log_all=True, simple=True)
         # clean up before proceeding with next build
-        run_cmd("./bjam --clean-all", log_all=True, simple=True)
+        run_cmd("./%s --clean-all" % bjamcmd, log_all=True, simple=True)
 
     def build_step(self):
         """Build Boost with bjam tool."""
 
+        if LooseVersion(self.version) >= LooseVersion("1.71.0"):
+            bjamcmd = 'b2'
+        else:
+            bjamcmd = 'bjam'
```

Comment:
Why not move this in the constructor and use `self.bjamcmd`, so you don't have to pass it to `self.build_boost_variant`?

Reply:
good idea.